receiver/couchdbreceiver/metrics.go

```diff
@@ -0,0 +1,199 @@
+// limitations under the License.
+
+package couchdbreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchdbreceiver"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchdbreceiver/internal/metadata"
+)
+
+func (c *couchdbScraper) recordCouchdbAverageRequestTimeDataPoint(now pdata.Timestamp, stats map[string]interface{}, errors scrapererror.ScrapeErrors) {
+	averageRequestTimeMetricKey := []string{"request_time", "value", "arithmetic_mean"}
+	averageRequestTimeValue, err := getValueFromBody(averageRequestTimeMetricKey, stats)
+	if err != nil {
+		errors.AddPartial(1, err)
+		return
+	}
+
+	parsedValue, err := c.parseFloat(averageRequestTimeValue)
+	if err != nil {
+		errors.AddPartial(1, err)
+		return
+	}
+	c.mb.RecordCouchdbAverageRequestTimeDataPoint(now, parsedValue)
+}
+
+func (c *couchdbScraper) recordCouchdbHttpdBulkRequestsDataPoint(now pdata.Timestamp, stats map[string]interface{}, errors scrapererror.ScrapeErrors) {
+	httpdBulkRequestsMetricKey := []string{"httpd", "bulk_requests", "value"}
+	httpdBulkRequestsMetricValue, err := getValueFromBody(httpdBulkRequestsMetricKey, stats)
+	if err != nil {
+		errors.AddPartial(1, err)
+		return
+	}
+
+	parsedValue, err := c.parseInt(httpdBulkRequestsMetricValue)
+	if err != nil {
+		errors.AddPartial(1, err)
+		return
+	}
+	c.mb.RecordCouchdbHttpdBulkRequestsDataPoint(now, parsedValue)
+}
+
+func (c *couchdbScraper) recordCouchdbHttpdRequestsDataPoint(now pdata.Timestamp, stats map[string]interface{}, errors scrapererror.ScrapeErrors) {
+	methods := []string{metadata.AttributeHTTPMethod.COPY, metadata.AttributeHTTPMethod.DELETE, metadata.AttributeHTTPMethod.GET, metadata.AttributeHTTPMethod.HEAD, metadata.AttributeHTTPMethod.OPTIONS, metadata.AttributeHTTPMethod.POST, metadata.AttributeHTTPMethod.PUT}
+	for _, method := range methods {
+		httpdRequestMethodKey := []string{"httpd_request_methods", method, "value"}
+		httpdRequestMethodValue, err := getValueFromBody(httpdRequestMethodKey, stats)
+		if err != nil {
+			errors.AddPartial(1, err)
+			continue
+		}
+
+		parsedValue, err := c.parseInt(httpdRequestMethodValue)
+		if err != nil {
+			errors.AddPartial(1, err)
+			continue
+		}
+		c.mb.RecordCouchdbHttpdRequestsDataPoint(now, parsedValue, method)
+	}
+}
+
+func (c *couchdbScraper) recordCouchdbHttpdResponsesDataPoint(now pdata.Timestamp, stats map[string]interface{}, errors scrapererror.ScrapeErrors) {
+	codes := []string{"200", "201", "202", "204", "206", "301", "302", "304", "400", "401", "403", "404", "405", "406", "409", "412", "413", "414", "415", "416", "417", "500", "501", "503"}
+	for _, code := range codes {
+		httpdResponsetCodeKey := []string{"httpd_status_codes", code, "value"}
+		httpdResponsetCodeValue, err := getValueFromBody(httpdResponsetCodeKey, stats)
+		if err != nil {
```

Comment:
Does the response necessarily contain a value for every status code? If it omits unused codes, then we'd have false errors here.

Reply:
Great, thanks for confirming.